src/tests/ui/anonymous-experiences/request.spec.ts

```diff
@@ -75,19 +75,32 @@ test.describe('Run query', () => {
     await page.locator('label:has-text("messages")').click();
     await page.evaluate(() => document.fonts.ready);
     expect(await page.screenshot({ clip: { x: 300, y: 0, width: 1920, height: 1080 } })).toMatchSnapshot();
-    await queryInputField.type('?sel');
+    await queryInputField.type('?$sel');
     await page.locator('label:has-text("$select")').click();
     await page.evaluate(() => document.fonts.ready);
     expect(await page.screenshot({ clip: { x: 300, y: 0, width: 1920, height: 1080 } })).toMatchSnapshot();
     await queryInputField.press('Tab');
     await queryInputField.press('Tab');
     await page.evaluate(() => document.fonts.ready);
     expect(await page.screenshot({ clip: { x: 300, y: 0, width: 1920, height: 1080 } })).toMatchSnapshot();
-
     // eslint-disable-next-line max-len
-    expect('input[aria-label="Query sample input"]:has-text("https://graph.microsoft.com/v1.0/me/messages?$select=id")').toBeDefined()
+    expect('input[aria-label="Query sample input"]:has-text("https://graph.microsoft.com/v1.0/me/messages?$select=id,changeKey")').toBeDefined()
   });
 
+  test('Tests query parameter addition on autocomplete', async () => {
+    const queryInputField = page.locator('[aria-label="Query sample input"]');
+    await queryInputField.click();
+    await page.evaluate(() => document.fonts.ready);
+    await queryInputField.fill('');
+    await queryInputField.fill('https://graph.microsoft.com/v1.0/me/messages');
+    await queryInputField.type('?sel');
+    await page.locator('label:has-text("$select")').click();
+    await page.evaluate(() => document.fonts.ready);
+    await queryInputField.press('Tab');
+    await queryInputField.press('Tab');
+    expect('input[aria-label="Query sample input"]:has-text("https://graph.microsoft.com/v1.0/me/messages?$select=id")').toBeDefined()
+  })
```

Comment:
With this kind of issue if you can write a test that models the reproduction steps and have it fail before you  implement the fix then you have both a fix and built in validation that the change did what your expected. 

I'm not clear on how this new test validates that typing in an $filter query works as expected?

Reply:
Sure. I have added a test for the $filter method to make sure that what is typed is what appears on the textbox